src/Processors/QueryPlan/QueryPlan.cpp

```diff
@@ -387,7 +387,15 @@ void QueryPlan::explainPlan(WriteBuffer & buffer, const ExplainPlanOptions & opt
 
 static void explainPipelineStep(IQueryPlanStep & step, IQueryPlanStep::FormatSettings & settings)
 {
-    settings.out << String(settings.offset, settings.indent_char) << "(" << step.getName() << ")\n";
+    // Add explicit description to the scan node of pipeline.
+    if ("ReadFromMergeTree" == step.getName())
+    {
+        settings.out << String(settings.offset, settings.indent_char) << "(" << step.getName() << " " << step.getStepDescription() << ")\n";
```

Comment:
I suppose it should be already added in other place, since description is set, isn't it?

Reply:
Sorry, I can't catch up with you. 
Do you mean the following operation in src/Processors/QueryPlan/ReadFromMergeTree.cpp: 
`setStepDescription(data.getStorageID().getFullNameNotQuoted());`

Reply:
When we `setStepDescription` it will already added to plan.
Not sure that we need description for pipeline, but if add it to pipeline as well, then do it in general way, not only for particular step.
Let's merge this PR with changes only for plan, I've already pushed corresponding commit into this PR.

Reply:
Thanks @vdimir ^_^